sktime/transformers/rocket.py

```diff
@@ -0,0 +1,239 @@
+        _num_channel_indices = num_channel_indices[i]
+
+        _weights = np.random.normal(0, 1, _num_channel_indices * _length)
+
+        b1 = a1 + (_num_channel_indices * _length)
+        b2 = a2 + _num_channel_indices
+
+        a3 = 0 # for weights (per channel)
+        for j in range(_num_channel_indices):
+            b3 = a3 + _length
+            _weights[a3:b3] = _weights[a3:b3] - _weights[a3:b3].mean()
+            a3 = b3
+
+        weights[a1:b1] = _weights
+
+        channel_indices[a2:b2] = np.random.choice(np.arange(0, num_channels), _num_channel_indices, replace = False)
+
+        biases[i] = np.random.uniform(-1, 1)
+
+        dilation = 2 ** np.random.uniform(0, np.log2((input_length - 1) / (_length - 1)))
+        dilation = np.int32(dilation)
+        dilations[i] = dilation
+
+        padding = ((_length - 1) * dilation) // 2 if np.random.randint(2) == 1 else 0
+        paddings[i] = padding
+
+        a1 = b1
+        a2 = b2
+
+    return weights, lengths, biases, dilations, paddings, num_channel_indices, channel_indices
+
+@njit(fastmath = True)
+def _apply_kernel_univariate(X, weights, length, bias, dilation, padding):
+
+    input_length = len(X)
+
+    output_length = (input_length + (2 * padding)) - ((length - 1) * dilation)
+
+    _ppv = 0
+    _max = np.NINF
+
+    end = (input_length + padding) - ((length - 1) * dilation)
+
+    for i in range(-padding, end):
+
+        _sum = bias
+
+        index = i
+
+        for j in range(length):
+
+            if index > -1 and index < input_length:
+
+                _sum = _sum + weights[j] * X[index]
+
+            index = index + dilation
+
+        if _sum > _max:
+            _max = _sum
+
+        if _sum > 0:
+            _ppv += 1
+
+    return _ppv / output_length, _max
+
+@njit(fastmath = True)
+def _apply_kernel_multivariate(X, weights, length, bias, dilation, padding, num_channel_indices, channel_indices):
+
+    num_channels, input_length = X.shape
+
+    output_length = (input_length + (2 * padding)) - ((length - 1) * dilation)
+
+    _ppv = 0
+    _max = np.NINF
+
+    end = (input_length + padding) - ((length - 1) * dilation)
+
+    for i in range(-padding, end):
+
+        _sum = bias
+
+        index = i
+
+        for j in range(length):
+
+            if index > -1 and index < input_length:
+
+                for k in range(num_channel_indices):
+
+                    _sum = _sum + weights[k, j] * X[channel_indices[k], index]
+
+            index = index + dilation
+
+        if _sum > _max:
+            _max = _sum
+
+        if _sum > 0:
+            _ppv += 1
+
+    return _ppv / output_length, _max
+
+@njit("float64[:,:](float64[:,:,:],Tuple((float64[::1],int32[:],float64[:],int32[:],int32[:],int32[:],int32[:])))", parallel = True, fastmath = True)
+def _apply_kernels(X, kernels):
+
+    weights, lengths, biases, dilations, paddings, num_channel_indices, channel_indices = kernels
+
+    num_examples, num_channels, _ = X.shape
```

Comment:
```suggestion
    n_instances, n_columns, n_timepoints = X.shape
```

Would be great to use our agreed terminology here (if my interpretation of your code is correct)

Reply:
Tests are passing now, so that should be the last bit before we can merge it! We're currently writing down a list of methods and their contributors, so I may ask you to add yourself with your method to the table.